### PR TITLE
Add application status

### LINKED ISSUE #87: Create application status command

### PR DESCRIPTION
**Pull request by mattisongjj**
This pull request creates a new application status command

Command format: status [COMPANY_INDEX] /ap [STATUS]

Closes #87

### CODE REVIEW

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -21,26 +22,27 @@
  */
 public class SampleDataUtil {
     public static final Bookmark DEFAULT_BOOKMARK = new Bookmark(false);
+    public static final ApplicationStatus DEFAULT_APPLICATION_STATUS = new ApplicationStatus("");
     public static Company[] getSampleCompanies() {
         return new Company[] {
             new Company(new Name("Google"), new Phone("00000001"), new Email("google@example.com"),
                     new Address("Blk 30 Geylang Street 29, #06-40"), new CareerPageUrl("www.google-careers.com"),
-                    getTagSet("Salary: Low", "WLB: High"), DEFAULT_BOOKMARK),
+                    DEFAULT_APPLICATION_STATUS, getTagSet("Salary: Low", "WLB: High"), DEFAULT_BOOKMARK),
```
Comment (vinceg4) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:30]: Perhaps can try adding test to cover these lines when you have the time.

**Review by vinceg4 — COMMENTED** [review]
Other than some missing coverage, LGTM!

File: src/main/java/seedu/address/logic/commands/ApplicationStatusCommand.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        Company companyToEdit = lastShownList.get(index.getZeroBased());
+        Company editedPerson = new Company(
+                companyToEdit.getName(), companyToEdit.getPhone(), companyToEdit.getEmail(),
+                companyToEdit.getAddress(), companyToEdit.getCareerPageUrl(),
+                companyToEdit.getApplicationStatus(), companyToEdit.getTags(), companyToEdit.getIsBookmark());
+
+        model.setCompany(companyToEdit, editedPerson);
+        model.updateFilteredCompanyList(Model.PREDICATE_SHOW_ALL_COMPANIES);
+
+        return new CommandResult(String.format(MESSAGE_ADD_STATUS_SUCCESS, Messages.format(companyToEdit)));
```
Comment (rynlwhh) [inline src/main/java/seedu/address/logic/commands/ApplicationStatusCommand.java:62]: Code is easy to read, however could have some test cases here to cover these lines of code.

File: src/main/java/seedu/address/storage/JsonAdaptedCompany.java [hunk 3]
```diff
@@ -124,6 +129,13 @@ public Company toModelType() throws IllegalValueException {
         }
         final CareerPageUrl modelCareerPageUrl = new CareerPageUrl(careerPageUrl);
 
+        if (applicationStatus == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    ApplicationStatus.class.getSimpleName()));
```
Comment (rynlwhh) [inline src/main/java/seedu/address/storage/JsonAdaptedCompany.java:134]: Would be good to have a test to cover these two lines!

File: src/main/java/seedu/address/model/company/ApplicationStatus.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            return false;
+        }
+
+        ApplicationStatus otherStatus = (ApplicationStatus) other;
+        return value.equals(otherStatus.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
```
Comment (rynlwhh) [inline src/main/java/seedu/address/model/company/ApplicationStatus.java:45]: Could consider a test case to cover these two lines

File: src/main/java/seedu/address/model/company/ApplicationStatus.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof ApplicationStatus)) {
+            return false;
```
Comment (rynlwhh) [inline src/main/java/seedu/address/model/company/ApplicationStatus.java:36]: Could implement a test case for this part of ApplicationStatus code also!

**Review by rynlwhh — COMMENTED** [review]
Other than a few lines of codes that needs to be covered by test cases, LGTM!

**Pull request merged**